### PR TITLE
feat: create deployment script

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: scripts/deployAll.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+    console.log(`\n== Deploying ${w3fName} ==`)
+
+    if (ipfsDeployment == undefined) {
+        console.log(`   * Skipping ${w3fName} deployment (no IPFS deployment found)`)
+    } else if (ipfsDeployment == gelatoDeployment) {
+        console.log(`   * Skipping ${w3fName} deployment (already deployed)`)
+    } else {
+        console.log(`   * Deploying ${w3fName}...`)
+        await deploymentLogic()
```
Comment (lucas-manuel) [inline scripts/deployAll.ts]: Pass in ipfsDeployment as a param to make it clearer

File: scripts/deployAll.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    // ********** XCHAIN ORACLE TICKER *********************************************************************************
+    // *****************************************************************************************************************
+    await deploy('xchain-oracle-ticker', async () => {
+        const dsNoteInterface = new ethers.utils.Interface(dsNoteAbi)
+
+        const { taskId: eventTaskId, tx: eventTx }: TaskTransaction = await mainnetAutomation.createBatchExecTask({
+            name: 'XChain DSR Oracle Ticker [Event Based]',
+            web3FunctionHash: ipfsDeployment,
+            web3FunctionArgs: {
+                maxDelta: '', // TODO: Add reasonable max rho delta
```
Comment (lucas-manuel) [inline scripts/deployAll.ts]: Need to add a delta here

Comment (lucas-manuel) [reply scripts/deployAll.ts]: For both

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: scripts/deployAll.ts (hunk 2 above)
Comment (barrutko) [reply scripts/deployAll.ts]: Done

File: scripts/deployAll.ts (hunk 1 above)
Comment (barrutko) [reply scripts/deployAll.ts]: Done

**Pull request merged**